src/crypto.rs

```diff
@@ -1,7 +1,165 @@
+use thiserror::Error;
 use ssz_rs::prelude::*;
+use rand::prelude::*;
+
+use blst::{min_pk as blst_core, BLST_ERROR};
+use crate::crypto::Error::SizeMismatch;
```

Comment:
one clippy complaint is this import which is not used. 

you can just remove it here. and in this file, we already have `Error` in scope so don't need to do anything special to use a variant

Reply:
fixed